auth/server/src/main/java/com/seagates3/parameter/validator/AccountParameterValidator.java

```diff
@@ -40,12 +40,30 @@
      */
     @Override
     public Boolean isValidCreateParams(Map<String, String> requestBody) {
-        if (!S3ParameterValidatorUtil.isValidEmail(requestBody.get("Email"))) {
-            return false;
+        int validParamsCount = 0;
+        if (S3ParameterValidatorUtil.isValidEmail(
+                requestBody.get("Email"))) {
+            validParamsCount++;
         }
 
-        return S3ParameterValidatorUtil.isValidName(
-                requestBody.get("AccountName"));
+        if(S3ParameterValidatorUtil.isValidName(
+                requestBody.get("AccountName"))) {
+            validParamsCount++;
+        }
+
+        String accessKey = requestBody.get("AccessKey");
+        String secretKey = requestBody.get("SecretKey");
+
+        if (accessKey != null || secretKey != null) {
```

Comment:
NUll check is repeating here and in isValidCustomAccesskeyKey method.

Reply:
Valid cases:

1. Both access key and secret key should not be provided (in this case system will generate the keys).
2. Both access key and secret key should be provided.

Invalid cases:

1. Only access key is given
2. Only secret key is given

To support all these cases we require the null checks at both places:

1. if (accessKey != null || secretKey != null) . This will check the Invalid cases.
2. isValidCustomAccessKey(accessKey) && isValidCustomSecretKey(secretKey). This will validate the Valid cases.
3. The null check in isValidCustomAccessKey and isValidCustomSecretKey will actually validate the keys.

Reply:
Why we are combining the 2 checks?
Why not to have separate condition check?
We making it complex

Reply:
Keep null check at both. As discussed implement rest

Reply:
Done